mantidproject/mantid
# Detector grouping options for OSIRISDiffractionReduction algorithm

### Description of work
This PR adds several algorithm properties to be used for detector grouping to the `OSIRISDiffractionReduction` algorithm. These grouping methods include 'Custom', 'Individual', 'File', 'Groups' and 'Workspace'.

The PR also deprecates the `MapFile` property in `ISISIndirectEnergyTransfer` and `ISISIndirectDiffractionReduction` and replaces it with a new property that does exactly the same thing called `GroupingFile`.

Fixes #36762

### To test:
1. Open `Indirect` -> `Diffraction` interface
2. Select `OSIRIS` instrument, and `diffonly`
3. Load `137793` as sample
4. Load `137713` as vanadium
5. Choose grouping Option 'All'. Click Run. The result workspace should have 1 histogram
5. Choose grouping Option 'Individual'. Click Run. The result workspace should have 959 histograms
5. Choose grouping Option 'Groups'. Choose 5 groups. Click Run. The result workspace should have 6 histograms (the last histogram contains the remainder of the division by 5
5. Choose grouping Option 'Custom'. Type in string `3-50,51+52,52-962`. Click Run. The result workspace should have 3 histograms







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
